include/eve/module/core/regular/impl/simd/x86/convert.hpp

```diff
@@ -10,8 +10,367 @@
+    else  if constexpr( c_i == float32x8  && c_o == int32x8  )          return _mm256_cvttps_epi32(v);
+    else  if constexpr( c_i == float32x16 && c_o == int32x16 )          return _mm512_cvttps_epi32(v);
+    else  if constexpr( c_i == float32x4  && c_o == uint32x4  && a512)  return _mm_cvttps_epu32(v);
+    else  if constexpr( c_i == float32x8  && c_o == uint32x8  && a512)  return _mm256_cvttps_epu32(v);
+    else  if constexpr( c_i == float32x16 && c_o == uint32x16)          return _mm512_cvttps_epu32(v);
+    else  if constexpr( sizeof(U) <= 2 )                                return convert(convert(v,i32_t{}), tgt);
+    else  return convert_impl(EVE_RETARGET(cpu_),v,tgt);
+  }
+
+  //================================================================================================
+  // convert: (u)int64 -> U
+  //================================================================================================
+  template<integral_scalar_value T, typename N, real_scalar_value U>
+  EVE_FORCEINLINE wide<U, N>  convert_impl( EVE_SUPPORTS(sse2_)
+                                          , wide<T, N> const& v, as<U> const &tgt
+                                          ) noexcept
+  requires (sizeof(T) == 8)
+  {
+    using enum category;
+    constexpr auto c_i    = categorize<wide<T, N>>();
+    constexpr auto c_o    = categorize<wide<U, N>>();
+    constexpr auto a512   = current_api >= avx512;
+    constexpr auto aavx2  = current_api >= avx2;
+    constexpr auto mi64x4 = match(c_i, int64x4, uint64x4);
+    constexpr auto mi64x8 = match(c_i, int64x8, uint64x8);
+    constexpr auto mo32x4 = match(c_o, int32x4, uint32x4);
+
+          if constexpr( c_i == int64x2  && c_o == float64x2 && a512)  return _mm_cvtepi64_pd(v);
+    else  if constexpr( c_i == uint64x2 && c_o == float64x2 && a512)  return _mm_cvtepu64_pd(v);
+    else  if constexpr( c_i == int64x2  && c_o == float32x4 && a512)  return _mm_cvtepi64_ps(v);
+    else  if constexpr( c_i == uint64x2 && c_o == float32x4 && a512)  return _mm_cvtepu64_ps(v);
+    else  if constexpr( N{} <= 2 && c_o == int16x8  && a512)          return _mm_cvtepi64_epi16(v);
+    else  if constexpr( N{} <= 2 && c_o == uint16x8 && a512)          return _mm_cvtepi64_epi16(v);
+    else  if constexpr( N{} <= 2 && c_o == int8x16  && a512)          return _mm_cvtepi64_epi8(v);
+    else  if constexpr( N{} <= 2 && c_o == uint8x16 && a512)          return _mm_cvtepi64_epi8(v);
+    else  if constexpr( N{} <= 2 && mo32x4 && a512 )                  return _mm_cvtepi64_epi32(v);
+    else  if constexpr( N{} <= 2 && mo32x4 )                          return convert_integers_shuffle(v,tgt);
+    else  if constexpr( N{} <= 2 && (c_o && integer_) )               return convert( convert(v, as<upgrade_t<U>>{}), tgt);
+    else  if constexpr( N{} <= 2 )                                    return convert_impl(EVE_RETARGET(cpu_),v,tgt);
```

Comment:
Why N <= 2? I don't get it. We ddon't do it like that.
Also - shuffle should be better than scalar.

Generally - N() <= 2 is weird.